plugins/wallabag.koplugin/main.lua

```diff
@@ -123,53 +131,58 @@ function Wallabag:init()
     if self.ui and self.ui.link then
         self.ui.link:addToExternalLinkDialog("25_wallabag", function(this, link_url)
             return {
-                text = _("Add to Wallabag"),
+                text = _("Add to wallabag"),
                 callback = function()
                     UIManager:close(this.external_link_dialog)
                     this.ui:handleEvent(Event:new("AddWallabagArticle", link_url))
                 end,
             }
         end)
     end
-end
+end -- Wallabag:init
 
+--- Add Wallabag to the Tools menu in both the file manager and the reader.
 function Wallabag:addToMainMenu(menu_items)
     menu_items.wallabag = {
-        text = _("Wallabag"),
+        text = _("Wallabag"), -- Leave uppercase for a clean menu
         sub_item_table = {
             {
-                text = _("Retrieve new articles from server"),
+                text_func = function()
+                    if self.auto_archive then
+                        return _("Synchronize articles with server")
+                    else
+                        return _("Download new articles from server")
+                    end
+                end,
                 callback = function()
-                    self.ui:handleEvent(Event:new("SynchronizeWallabag"))
+                    self.ui:handleEvent(Event:new("DownloadArticles"))
                 end,
             },
             {
-                text = _("Delete finished articles remotely"),
+                text = T(
+                    _("Upload queue of locally added articles to server (%1)"),
+                    #self.upload_queue
+                ),
```

Comment:
Also not a fan of most changes of this type of reduction in clarity and greppability. :-)